File: mobile-ios/App/Models/SessionPayloads.swift
import Foundation

struct SessionOfferPayload: Codable {
    let version: String
    let role: String
    let sessionId: String
    let senderDeviceName: String
    let senderPubKeyFingerprint: String
    let offerSdp: String
    let expiresAtUnixMs: Int64

    init(
        sessionId: String,
        senderDeviceName: String,
        senderPubKeyFingerprint: String,
        offerSdp: String,
        expiresAtUnixMs: Int64
    ) {
        self.version = "1"
        self.role = "sender"
        self.sessionId = sessionId
        self.senderDeviceName = senderDeviceName
        self.senderPubKeyFingerprint = senderPubKeyFingerprint
        self.offerSdp = offerSdp
        self.expiresAtUnixMs = expiresAtUnixMs
    }
}

struct SessionAnswerPayload: Codable {
    let version: String
    let role: String
    let sessionId: String
    let receiverDeviceName: String
    let receiverPubKeyFingerprint: String
    let answerSdp: String
    let expiresAtUnixMs: Int64

    init(
        sessionId: String,
        receiverDeviceName: String,
        receiverPubKeyFingerprint: String,
        answerSdp: String,
        expiresAtUnixMs: Int64
    ) {
        self.version = "1"
        self.role = "receiver"
        self.sessionId = sessionId
        self.receiverDeviceName = receiverDeviceName
        self.receiverPubKeyFingerprint = receiverPubKeyFingerprint
        self.answerSdp = answerSdp
        self.expiresAtUnixMs = expiresAtUnixMs
    }
}

File: mobile-android/app/src/main/java/com/example/p2paudio/model/SessionPayloads.kt
package com.example.p2paudio.model

import kotlinx.serialization.Serializable

@Serializable
data class PairingInitPayload(
    val version: String = "2",
    val phase: String = "init",
    val sessionId: String,
    val senderDeviceName: String,
    val senderPubKeyFingerprint: String,
    val offerSdp: String,
    val expiresAtUnixMs: Long
)

@Serializable
data class PairingConfirmPayload(
    val version: String = "2",
    val phase: String = "confirm",
    val sessionId: String,
    val receiverDeviceName: String,
    val receiverPubKeyFingerprint: String,
    val answerSdp: String,
    val expiresAtUnixMs: Long
)

@Serializable
data class UdpInitPayload(
    val version: String = "2",
    val phase: String = "udp_init",
    val transport: String = "udp_opus",
    val sessionId: String,
    val senderDeviceName: String,
    val expiresAtUnixMs: Long
)

@Serializable
data class UdpConfirmPayload(
    val version: String = "2",
    val phase: String = "udp_confirm",
    val transport: String = "udp_opus",
    val sessionId: String,
    val receiverDeviceName: String,
    val receiverPort: Int,
    val expiresAtUnixMs: Long
)
